### ISSUE TITLE
MX record priority error

### ISSUE THREAD

**Issue by khromov** [jungaretti/bacon#22]
👋 I was using this script to migrate a domain but ran into this error mid-way:

```
Error: couldn't create record: unmarshaling JSON body: invalid character '<' looking for beginning of value Usage:
```

When checking in the Porkbun admin it seemed to be a faulty MX record that got created but had priority 0. After manually changing the priority from PB admin panel the script could proceed.

Here is a snippet from my yaml:

```
  - type: MX
    host: "x.domain.com"
    content: mxa.eu.mailgun.org.
    ttl: 600
    priority: 10
```

As an aside, for some reason Bacon can't seem to detect that this particular DNS entry is already created:

```
  - type: TXT
    host: "bun-verify.khromov.se"
    content: "127d8e80396db8b58b2f236991f5e715c5431df0763ccaef0d2edd516220308a"
    ttl: 300
```

Every time i re-run it it says:

```
./bacon deploy khromov.se.yaml --delete --create
Deleting 1 records...
- {344101976 bun-verify.khromov.se TXT 127d8e80396db8b58b2f236991f5e715c5431df0763ccaef0d2edd516220308a 600 0 }
Creating 1 records...
- {bun-verify.khromov.se TXT 0xc0000b9450 127d8e80396db8b58b2f236991f5e715c5431df0763ccaef0d2edd516220308a}
Deployment complete!
```

Thanks for a great tool!

**Comment by jungaretti** [comment]
Bacon has trouble de-serializing and comparing numerical properties such as TTL and priority. I suspect that both of these problems relate to flakiness in Bacon's conversion to/from `string`

**Comment by jungaretti** [comment]
I recently discovered that the error `Error: couldn't create record: unmarshaling JSON body: invalid character '<' looking for beginning of value` is a red herring.

Before #42, Bacon did not check for a `2XX` status code before parsing the response body. Last night, I discovered that Porkbun sometimes returns a `503` if you send too many requests too quickly. I suspect that this `503` behavior is the true cause of your issue here.

I'm going to close this issue as not planned. If anyone sees this error again, let's re-activate this issue or create a new one with the newly improved error message.

**Comment by jungaretti** [comment]
Your `bun-verify.khromov.se` domain has an invalid TTL. Porkbun requires that TTL be at least 600 seconds. When Bacon sends a request for your record with TTL `300`, Porkbun silently increases its TTL to `600`. So the record in Porkbun is never equivalent to the record in your config file. That's why Bacon repeatedly deploys it.

This is fixed in [v2.0](https://github.com/jungaretti/bacon/releases/tag/v2.0) where a new error message will clearly indicate that the minimum TTL value is 600.
